packages/next/server/config-shared.ts

```diff
@@ -58,126 +172,215 @@ export type NextConfig = { [key: string]: any } & {
         fallback: Rewrite[]
       }
   >
+
+  /**
+   * Redirects allow you to redirect an incoming request path to a different destination path.
+   *
+   * @see [Redirects configuration documentations](https://nextjs.org/docs/api-reference/next.config.js/redirects)
+   */
   redirects?: () => Promise<Redirect[]>
 
   /**
    * @deprecated This option has been removed as webpack 5 is now default
+   * @see [Next.js webpack 5 documentation](https://nextjs.org/docs/messages/webpack5) for upgrading guidance.
    */
   webpack5?: false
+
+  /**
+   * @see [Moment.js locales excluded by default](https://nextjs.org/docs/upgrading#momentjs-locales-excluded-by-default)
+   */
   excludeDefaultMomentLocales?: boolean
 
-  webpack?:
-    | ((
-        config: any,
-        context: {
-          dir: string
-          dev: boolean
-          isServer: boolean
-          buildId: string
-          config: NextConfigComplete
-          defaultLoaders: { babel: any }
-          totalPages: number
-          webpack: any
-        }
-      ) => any)
-    | null
+  /**
+   * Before continuing to add custom webpack configuration to your application make sure Next.js doesn't already support your use-case
+   *
+   * @see [Custom Webpack Config documentation](https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config)
+   */
+  webpack?: NextJsWebpackConfig | null
 
+  /**
+   * By default Next.js will redirect urls with trailing slashes to their counterpart without a trailing slash.
```

Comment:
This description should explain the difference between `true` and `false` and then list what the default value is.